File: skills/tests/resources/IntegerMultiplyBy2UDF.scala
/*
 * SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

package com.udf

class IntegerMultiplyBy2UDF extends Function1[Integer, Integer] with Serializable {
  override def apply(value: Integer): Integer = {
    if (value == null) null else value * 2
  }
}

File: skills/tests/resources/IntegerMultiplyBy2UDF.java
/*
 * SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

package com.udf;

import org.apache.spark.sql.api.java.UDF1;

public class IntegerMultiplyBy2UDF implements UDF1<Integer, Integer> {
  @Override
  public Integer call(Integer value) {
    return value == null ? null : value * 2;
  }
}
